services/core-web/common/actionCreators/permitActionCreator.ts

```diff
@@ -0,0 +1,500 @@
+        dispatch(success(reducerTypes.GET_PERMIT_AMENDMENT));
+        return response.data;
+      })
+      .catch((err) => {
+        dispatch(error(reducerTypes.GET_PERMIT_AMENDMENT));
+        throw new Error(err);
+      })
+      .finally(() => dispatch(hideLoading()));
+  };
+
+export const removePermitAmendmentDocument =
+  (mineGuid: string, permitGuid: string, permitAmdendmentGuid: string, documentGuid: string) =>
+  (dispatch) => {
+    dispatch(request(reducerTypes.UPDATE_PERMIT_AMENDMENT_DOCUMENT));
+    dispatch(showLoading());
+    return CustomAxios()
+      .delete(
+        `${ENVIRONMENT.apiUrl}${API.PERMIT_AMENDMENT_DOCUMENT(
+          mineGuid,
+          permitGuid,
+          permitAmdendmentGuid,
+          documentGuid
+        )}`,
+        createRequestHeader()
+      )
+      .then((response) => {
+        notification.success({
+          message: `Successfully removed attached document`,
+          duration: 10,
+        });
+        dispatch(success(reducerTypes.UPDATE_PERMIT_AMENDMENT_DOCUMENT));
+        return response;
+      })
+      .catch((err) => {
+        dispatch(error(reducerTypes.UPDATE_PERMIT_AMENDMENT_DOCUMENT));
+        throw new Error(err);
+      })
+      .finally(() => dispatch(hideLoading()));
+  };
+
+export const deletePermit = (mineGuid: string, permitGuid: string) => (dispatch) => {
+  dispatch(request(reducerTypes.DELETE_PERMIT));
+  dispatch(showLoading());
+  return CustomAxios()
+    .delete(
+      `${ENVIRONMENT.apiUrl}${API.PERMIT_DELETE(mineGuid, permitGuid)}`,
+      createRequestHeader()
+    )
+    .then((response) => {
+      notification.success({
+        message: "Successfully deleted permit and all related permit amendments and documents.",
+        duration: 10,
+      });
+      dispatch(success(reducerTypes.DELETE_PERMIT));
+      return response;
+    })
+    .catch((err) => {
+      dispatch(error(reducerTypes.DELETE_PERMIT));
+      throw new Error(err);
+    })
+    .finally(() => dispatch(hideLoading()));
+};
+
+export const deletePermitAmendment =
+  (mineGuid: string, permitGuid: string, permitAmdendmentGuid: string) => (dispatch) => {
+    dispatch(request(reducerTypes.DELETE_PERMIT_AMENDMENT));
+    dispatch(showLoading());
+    return CustomAxios()
+      .delete(
+        `${ENVIRONMENT.apiUrl}${API.PERMIT_AMENDMENT(mineGuid, permitGuid, permitAmdendmentGuid)}`,
+        createRequestHeader()
+      )
+      .then((response) => {
+        notification.success({
+          message: "Successfully deleted permit amendment and all related documents.",
+          duration: 10,
+        });
+        dispatch(success(reducerTypes.DELETE_PERMIT_AMENDMENT));
+        return response;
+      })
+      .catch((err) => {
+        dispatch(error(reducerTypes.DELETE_PERMIT_AMENDMENT));
+        throw new Error(err);
+      })
+      .finally(() => dispatch(hideLoading()));
+  };
+
+export const fetchPermitConditions = (permitAmdendmentGuid) => (dispatch) => {
```

Comment:
type for permitAmendmentGuid is missing here.

Reply:
Ahh Just wanted to mention some of the functions in permitActionCreator weren't converted. I just converted the functions related to MinePermitInfo component. There are some functions related to other components that haven't been converted yet, my thinking was when those components get converted the functions here can be as well.

Reply:
Oh okay, cool. I will just ignore them.